controllers/operatorconfig/operatorconfig_controller.go

```diff
@@ -47,14 +58,111 @@ type Reconciler struct {
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			klog.Infof("Fetching OperatorConfig: %s", operator.OperatorConfig)
+			config := &operatorv1alpha1.OperatorConfig{}
+			if err := r.Client.Get(ctx, types.NamespacedName{
+				Name:      operator.OperatorConfig,
+				Namespace: registry.Namespace,
+			}, config); err != nil {
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+			serviceConfig := config.GetConfigForOperator(operator.Name)
+			if serviceConfig == nil {
+				klog.Infof("OperatorConfig: %s, does not have configuration for operator: %s", operator.OperatorConfig, operator.Name)
+				return ctrl.Result{}, nil
+			}
+
+			copyToCast, err := deepcopy.Anything(csv)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			csvToUpdate := copyToCast.(*olmv1alpha1.ClusterServiceVersion)
+			klog.Infof("Applying OperatorConfig: %s to Operator: %s via CSV: %s, %s", operator.OperatorConfig, operator.Name, csv.Name, csv.Namespace)
+			return r.configCsv(ctx, csvToUpdate, serviceConfig)
+		}
+	}
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) configCsv(ctx context.Context, csv *olmv1alpha1.ClusterServiceVersion, config *operatorv1alpha1.ServiceOperatorConfig) (ctrl.Result, error) {
+	if config.Replicas != nil {
+		csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Spec.Replicas = config.Replicas
+	}
+	if config.Affinity != nil {
+		csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Spec.Template.Spec.Affinity = config.Affinity
+	}
+	if config.TopologySpreadConstraints != nil {
+		csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Spec.Template.Spec.TopologySpreadConstraints = config.TopologySpreadConstraints
+	}
+	if err := r.Client.Update(ctx, csv); err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) requestsFromMapFunc(ctx context.Context) handler.MapFunc {
+	return func(object client.Object) []reconcile.Request {
+		requests := []reconcile.Request{}
+
+		operandRequests, _ := r.ListOperandRequests(ctx, nil)
+		for _, req := range operandRequests.Items {
+			r := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: req.Namespace,
+					Name:      req.Name,
+				},
+			}
+			requests = append(requests, r)
+		}
+		return requests
+	}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ctx := context.Background()
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&operatorv1alpha1.OperatorConfig{}).
+		For(&operatorv1alpha1.OperandRequest{}).
+		Watches(&source.Kind{Type: &operatorv1alpha1.OperatorConfig{}}, handler.EnqueueRequestsFromMapFunc(r.requestsFromMapFunc(ctx)), builder.WithPredicates(predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				return true
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				// Evaluates to false if the object has been confirmed deleted.
+				return !e.DeleteStateUnknown
+			},
```

Comment:
I am wondering if a update event on OperandRequest should trigger the reconciliation as well?

Reply:
The predicates here are for events to the `OperatorConfig` CR not, `OperandRequest`. And if no predicate is specified for an event, then those events will be reconciled